### DISCUSSION TITLE
anime.js and swup

### DISCUSSION THREAD

**Discussion by Josedev24** [swup/swup#898] [Help]
### Describe the problem 🧐

I created two web pages in html, css and js. To create some effects on the text and elements on the pages I used the animate.js library.

I added the lenis.js library to have smooth scrolling, and everything works fine.

Wanting to insert transitions between the two pages, I used the swup.js library, but the problem is that transitions work well, but animated text with the animate library doesn't work.

Here's the pages:

[swup.zip]  — link removed
[swup.zip]  — link removed

### Describe the propsed solution 😎

How could I solve the problem? Maybe delaying animate.js animations immediately after the page transition made with swup or a similar solution.

### Alternatives considered 🤔

Use another method for page transitions

### How important is this feature to you? 🧭

Nice to have

### Checked all these? 📚

- [X] I have read and searched the [official docs](https://swup.js.org)
- [X] I have checked [discussions](https://github.com/orgs/swup/discussions) and [existing issues](https://github.com/issues?q=is:issue+org:swup) for related problems
- [X] I agree to follow this project's [Code of Conduct](https://github.com/swup/.github/blob/master/CODE_OF_CONDUCT.md)
- [X] I have provided all necessary information to the best of my knowledge

**Comment by daun** [comment]
Hi! I've converted this to a discussion as it's more of an integration question than an issue with swup.

And a little note on the side: nobody in their right mind will click on and download zip files from a public comment :) If you want help by other people, consider putting a public version of your project online, e.g. on CodeSandbox, Replit, Glitch, etc.

**Comment by daun** [comment]
Your issue might be related to (re)running javascript on each page visit, e.g. the anime.js init scripts mentioned. Feel free to check out the official docs on [Reloading JavaScript](https://swup.js.org/getting-started/reloading-javascript/) and report back.

**Comment by Josedev24** [comment]
Good morning and thank you for your response and suggestion. Yes, in fact it involves loading the anime.js library immediately after the page transition made with swup.js
Unfortunately I'm not familiar with coding, however I will try to make some attempts, if anyone is more familiar than me, please help me, thanks.

**Reply by Josedev24** [reply]
After a few weeks and after repeated attempts I was unable to resolve. Is there someone who can help me? Thank you

**Reply by daun** [reply]
To increase your chances of someone taking a look at this, you might consider uploading the code in question to a public repl like CodeSandbox or Replit. That will make it much easier for people to contribute than the currently linked zip files.

**Reply by Josedev24** [reply]
Thanks so much for the advice, I thought I already included it....can I attach a zip file?
[swup.zip] — link removed
[swup.zip] — link removed

**Reply by daun** [reply]
Downloading zip files from a public website is a security risk most people will not accept, especially for helping a stranger. Who knows what's in there. Uploading them to a public repl is the accepted way to go here :)

**Reply by Josedev24** [reply]
Good afternoon, I apologize for posting zip files, I'm not very practical, thanks for the advice, I hope I can do it. Please tell me an online service? Thank you

**Reply by hirasso** [reply]
CodeSandbox or Replit

**Reply by Josedev24** [reply]
Thank you for reply and advice. I'm trying to use codesandbox.io to share the files, it's not so intuitive :-(
Hope to do it well....

**Reply by Josedev24** [reply]
Here the URL with all the files:

https://codesandbox.io/p/sandbox/swup-js-and-anime-js-conflict-3m5y34

Thanks for attention

**Reply by Josedev24** [reply]
I created the project online:
two html pages where before the closing body tag I added the swup library (version 3) and anime (contained in the "vendors.min.js" file).
Page transition with swup works, but it doesn't reload the animations from the "anime.js" library.

How do I write a function where the "anime.js" animations are executed immediately after the "swup.js" library transaction?

I tried with the documentation on the site "https://swup.js.org/" but they are incomplete and don't work... it's starting to get frustrating, can someone with experience help me? Thank you

Here the replit URL:

https://replit.com/@gbisb75/Swupjs-and-animejs?v=1

**Reply by daun** [reply]
I've had a quick look at the project — I'm sorry to say but looking over those 3000 lines of javascript is just too much to ask of anybody unless you're paying them. Consider stripping everything that's unrelated to the problem you're experiencing to land at a [minimal reproduction](https://swup.js.org/other/minimal-reproductions/) or finding a freelancer to work with on this.

**Reply by daun** [reply]
If you have specific suggestions on what's missing in the official docs, we're happy to hear them!

**Reply by daun** [reply]
The "official" way of re-running code after each swup page transition is by [registering a hook handler](https://swup.js.org/hooks/) for the page view hook. See the example below.

```js
swup.hooks.on('page:view', (visit) => {
  /* run anime.js animations */
});
```

**Reply by Josedev24** [reply]
First of all, thank you for your messages. This problem is driving me crazy ;-) but in any case I'm trying to solve it step by step, looking for online resources, forums etc etc.

I believe, despite not knowing anything about js programming but going by "intuition" that the whole problem is just writing a few lines of code in the js function at the end of the document so that (once swup finishes loading the destination page) so that the animation of the "anime.js" library starts. 
I have already partially solved this problem: using the plugin indicated in the swup documentation pages, it would be enough to write the relevant js code for the anime.js library, ensuring that all the elements ("target") of the library are loaded sequentially page, as I indicate below:

<script src="https://unpkg.com/swup@3"></script>
        <script>
            const swup = new Swup();

            // Run once when page loads
            if (document.readyState === 'complete') {
            init();
            } else {
            document.addEventListener('DOMContentLoaded', () => init());
            }

            // Run after every additional navigation by swup
            swup.on('pageView', () => {
                
                var tl = anime.timeline({
                    translateY: [30, 0],
                    opacity: [0, 1],
                    duration: 800,
                    easing: 'easeOutQuad',
                    delay: 500
                });

                // Add children
                tl
                .add({
                    targets: '.title',
                    translateY: [30, 0],
                    opacity: [0, 1],
                    duration: 400,
                    delay: 200,
                    staggervalue: 150
                    
                }) ......and so on....
            });
        </script>
        
        This method works, but it has a problem: the animations start all together and not when the visitor reaches the viewport. 
Instead, in the file I included, the animations are linked directly in the html code in the different elements to be animated: this makes insertion where needed easier and faster and allows the animation to be displayed in the viewport.

Finally: in your message you show me the official code from the swup documentation:

swup.hooks.on('page:view', (visit) => {
  /* run anime.js animations */
});

Here...where it says "run anime.js"
All I need is a code that says it allows all the animations on the page to take place, without having to specify each target...
Sorry, it seems a bit complex, but I try to explain myself as clearly as possible, my English is not excellent :-)

Could you ask the gentlemen who developed swup to post a complete example with anime.js?

Thanks again for your support
Giuseppe

**Reply by hirasso** [reply]
Guiseppe, you are asking too much. You also ignored the request for a minimal reproduction. I‘ll close this as that kind of behaviour is not acceptable here.

**Reply by Josedev24** [reply]
Honestly, I went into the explanation in order to provide as many details as possible so that someone could help me more easily. I didn't think this violated the community rules. Thank you all the same.
